lib/services/ios-project-service.ts

```diff
@@ -980,28 +977,11 @@ We will now place an empty obsolete compatability white screen LauncScreen.xib f
 		this.$cocoapodsService.removePodfileFromProject(pluginData.name, this.$cocoapodsService.getPluginPodfilePath(pluginData), projectData, projectRoot);
 	}
 
-	public async afterPrepareAllPlugins(projectData: IProjectData): Promise<void> {
-		await this.installPodsIfAny(projectData);
-	}
-
-	public async installPodsIfAny(projectData: IProjectData): Promise<void> {
+	public async handleNativeDependenciesChange(projectData: IProjectData): Promise<void> {
 		const projectRoot = this.getPlatformData(projectData).projectRoot;
-		const mainPodfilePath = path.join(projectData.appResourcesDirectoryPath, this.getPlatformData(projectData).normalizedPlatformName, constants.PODFILE_NAME);
-		if (this.$fs.exists(this.$cocoapodsService.getProjectPodfilePath(projectRoot)) || this.$fs.exists(mainPodfilePath)) {
-			const xcodeProjPath = this.getXcodeprojPath(projectData);
-			const xcuserDataPath = path.join(xcodeProjPath, "xcuserdata");
-			const sharedDataPath = path.join(xcodeProjPath, "xcshareddata");
-
-			if (!this.$fs.exists(xcuserDataPath) && !this.$fs.exists(sharedDataPath)) {
-				this.$logger.info("Creating project scheme...");
-				await this.checkIfXcodeprojIsRequired();
-
-				const createSchemeRubyScript = `ruby -e "require 'xcodeproj'; xcproj = Xcodeproj::Project.open('${projectData.projectName}.xcodeproj'); xcproj.recreate_user_schemes; xcproj.save"`;
-				await this.$childProcess.exec(createSchemeRubyScript, { cwd: this.getPlatformData(projectData).projectRoot });
-			}
-
-			await this.$cocoapodsService.applyPodfileToProject(constants.NS_BASE_PODFILE, mainPodfilePath, projectData, this.getPlatformData(projectData).projectRoot);
-
+		const xcodeProjPath = this.getXcodeprojPath(projectData);
+		const projectPodfilePath = this.$cocoapodsService.getProjectPodfilePath(projectRoot);
+		if (this.$fs.exists(projectPodfilePath)) {
```

Comment:
This maybe should be executed also if `this.$fs.exists(mainPodfilePath)` is true.
```typescript
if (this.$fs.exists(projectPodfilePath) || this.$fs.exists(mainPodfilePath)) {
```

Reply:
Actually we don't have `mainPodfilePath` variable in this method and we don't need it.
The idea here is `pod install` command to be executed on pod file in platforms folder. This pod file is the output from merge of all pod files from `node_modules` and the pod file from `App_Resources` if such. So if we have pod file in `App_Resources` (`mainPodfilePath`) it will be merged in pod file in platforms folder. If we don't have pod file in `App_Resources`, `pod install` will be executed only when pod file in platforms dir exists e.g the app contains some plugins with pod files.